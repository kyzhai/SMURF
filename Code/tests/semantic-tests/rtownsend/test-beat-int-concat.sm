/* Should be able to concatenate list of ints to list of beats if ints are all powers of 2 */

x = [2,4,8] ++ [2.,4.]
main = []
