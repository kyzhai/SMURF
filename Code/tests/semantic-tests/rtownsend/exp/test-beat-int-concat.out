Program: 
	SVardef: 
		ID: x  :: [Beat]

	[2, 4, 8] ++ [2., 4.]
	SMain: []
Global Scope: 
	ID: main  :: []
	ID: x  :: [Beat]
	ID: print x :: a -> a
	ID: random  :: Int
	ID: head h : t :: [a] -> a
	ID: tail h : t :: [a] -> [a]

